test/tests/bootstrap/test_redfish10_os_bootstrap_parallel.py

```diff
@@ -0,0 +1,196 @@
+          {
+              "localPath": "/CentOS/6.5",
+              "remotePath": "/",
+              "server": "http://os-mirror-server/centos/6.5/os/x86_64"
+          },
+          {
+              "localPath": "/CentOS/7.0",
+              "remotePath": "/",
+              "server": "http://os-mirror-server/centos/7/os/x86_64"
+          },
+          {
+              "localPath": "/RHEL/7.0",
+              "remotePath": "/",
+              "server": "http://os-mirror-server/rhel/7.0/os/x86_64"
+          }
+      ],
+
+For each OS type, the "localpath" string must conform to the convention above.
+The "server' URL points to the location of the OS executables in an image repository.
+
+'''
+
+import os
+import sys
+import subprocess
+
+# set path to common libraries
+sys.path.append(subprocess.check_output("git rev-parse --show-toplevel", shell=True).rstrip("\n") + "/test/common")
+import fit_common
+
+import flogging
+log = flogging.get_loggers()
+
+# This gets the list of nodes
+NODECATALOG = fit_common.node_select()
+
+# dict containing bootstrap workflow IDs and states
+NODE_STATUS = {}
+
+# download RackHD config from host
+rackhdconfig = fit_common.rackhdapi('/api/2.0/config')['json']
+rackhdhost = "http://" + str(rackhdconfig['apiServerAddress']) + ":" + str(rackhdconfig['apiServerPort'])
+
+
+# this routine polls a task ID for completion
+def wait_for_task_complete(taskid, retries=60):
+    for dummy in range(0, retries):
+        result = fit_common.rackhdapi(taskid)
+        if result['status'] != 200:
+            log.error(" " + result['text'])
+            return False
+        if result['json']['TaskState'] == 'Running' or result['json']['TaskState'] == 'Pending':
+            if fit_common.VERBOSITY >= 2:
+                print "OS Install workflow state: {}".format(result['json']['TaskState'])
+            fit_common.time.sleep(30)
+        elif result['json']['TaskState'] == 'Completed':
+            if fit_common.VERBOSITY >= 2:
+                print "OS Install workflow state: {}".format(result['json']['TaskState'])
+            log.info_5(" " + result['text'])
+            return True
+        else:
+            log.error(" " + result['text'])
+            return False
+    log.error(" Workflow Timeout: " + result['text'])
+    return False
+
+# helper routine for selecting OS image path by matching proxy path
+def proxySelect(tag):
+    for entry in rackhdconfig['httpProxies']:
+        if tag in entry['localPath']:
+            return entry['localPath']
+    return ''
+
+# helper routine to return the task ID associated with the running bootstrap workflow
+def node_taskid(osname, version):
+    for entry in NODE_STATUS:
+        if NODE_STATUS[entry]['os'] == osname and NODE_STATUS[entry]['version'] == version:
+            return NODE_STATUS[entry]['id']
+    return ""
+
+# run all bootstraps if regression group is specified or no group is specified
+if "regression" in sys.argv or "-a" not in sys.argv:
```

Comment:
@gpaulos 
Hopefully this helps, especially now that I figured out how to actually get a code snippet in here!
```python
import unittest

class test_thing(unittest.TestCase):
    def setUp(self):
        if "regression" not in sys.argv or some_other_check_for_if_we_should_skip_all_these:
            raise unittest.SkipTest('Skipping because regression or all test not selected')
```

You might also be able to directly use attr inside setup, kind of like this: (this is totally theoretical, however. I have not tried this one yet. I have basically on the above one though)
```python
import unittest

class test_thing(unittest.TestCase):
    def setUp(self):
        if not attr(all=True, regression=True):
            raise unittest.SkipTest('Skipping because regression or all test not selected')
```